appserver/web/web-glue/src/main/java/org/glassfish/web/deployment/descriptor/UserDataConstraintImpl.java

```diff
@@ -99,18 +100,18 @@ public void setTransportGuarantee(String transportGuarantee) {
 		    && !UserDataConstraint.CONFIDENTIAL_TRANSPORT.equals(transportGuarantee)) {
 		throw new IllegalArgumentException(localStrings.getLocalString(
 									       "web.deployment.exceptiontransportguarentee",
-									       "{0} is not a valid transport guarantee", new Object[] {transportGuarantee}));  
+									       "{0} is not a valid transport guarantee", new Object[] {transportGuarantee}));
 	    }
 	}
 	this.transportGuarantee = transportGuarantee;
     }
-    
+
     /**
     * Returns a formatted String of my state.
     */
-    public void print(StringBuffer toStringBuffer) {
-	toStringBuffer.append("UserDataConstraint ");
-	toStringBuffer.append(" description ").append(super.getDescription());
-	toStringBuffer.append(" transportGuarantee ").append(getTransportGuarantee());
+    public void print(StringBuilder toStringBuilder) {
```

Comment:
Possible thread issues here

Reply:
This print methods only usage is from the parents `DynamicAttributesDescriptor.toString` method . This creates a new `StringBuilder` for every call. This is thread-safe.